Use Codestarts as default project generation tool

ia3andy opened the pull request:
- Codestarts which only works alone are called "Singleton"
- Added checks to avoid conflict between Singelton codestarts
- Make codestart the default codegen and add a legacyCodegen flag
- Use RESTEasy example as default (commandmode example is not selectable for now.. until a picocli https://github.com/quarkusio/quarkus/issues/12633 example?)
- Update the guides with `-DnoExamples` when when relevent
- Update the tests

Fixes #12113

ia3andy commented:
cc @maxandersen that gives you an idea of the PR

gastaldi commented on independent-projects/tools/codestarts/src/main/java/io/quarkus/devtools/codestarts/CodestartProjectInput.java:
```diff
@@ -2,14 +2,13 @@
 
 public class CodestartProjectInput {
-    private final Collection<AppArtifactKey> dependencies;
+    private final Collection<String> dependencies;
```
Why not use `AppArtifactKey` or `AppArtifactCoords`?

ia3andy replied:
Because this field can accept both (ga or gav), `AppArtifactCoords` doesn't extend `AppArtifactKey` and `AppArtifactCoords` is not meant to be used without version.

aloubyansky commented:
> Update the guides for compat by adding resteasy when relevent

What kind of compatibility? I understand that the plain resteasy extension isn't added in many cases now. My question is what kind of issue does it create?

ia3andy commented:
> > Update the guides for compat by adding resteasy when relevent
> 
> What kind of compatibility? I understand that the plain resteasy extension isn't added in many cases now. My question is what kind of issue does it create?

Some of our guides instruction are based on the resteasy example, thus without it in the command extensions, it is not generated. Soon, we will have example for most extensions, then we will have to remove resteasy, and the example will directly be the appropriate one.. (like we just did with spring-web)

ia3andy commented:
I am waiting for the result of this discussion to make the decided changes: https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/quarkus-dev/-j7C9UqYbF4/qyiTzQU5AAAJ

ia3andy committed:
Codestarts as default codegen

- Codestarts which only works alone are called "Singleton"
- Added checks to avoid conflict between Singelton codestarts
- Make codestart the default codegen and add a legacyCodegen flag
- Use RESTEasy example as default (commandmode example is not selectable for now.. until a picocli example?)
- Update the guides with `-DnoExamples` when when relevent
- Update the tests

Fixes #12113

ia3andy commented:
Sorry @aloubyansky this will need to be reviewed again. Since your approval, I've set resteasy as default and updated the docs and tests accordingly..

gastaldi reviewed (approved):
LGTM
